examples/customsortfiltermodel/main.go

```diff
@@ -0,0 +1,93 @@
+	treeView.SetAlternatingRowColors(true)
+	treeView.SetSortingEnabled(true)
+	treeView.SetColumnWidth(0, 300)
+	treeView.SetContentsMargins(0, 0, 0, 0)
+	treeView.Header().SetDefaultSectionSize(200)
+	treeView.Header().SetStretchLastSection(true)
+	treeView.SetSelectionMode(qt.QAbstractItemView__ExtendedSelection)
+
+	return treeView
+}
+
+func PopulateData(rootItem *TreeItem) {
+	cellData := make([]*TreeCell, 0)
+	cellData = append(cellData, NewTreeCell(qt.DisplayRole, *qt.NewQVariant11("Big File")))
+	cellData = append(cellData, NewTreeCell(qt.DisplayRole, *qt.NewQVariant11("10.7 MiB")))
+	item := NewTreeItem(cellData, rootItem, false)
+	err := item.SetData(1, qt.NewQVariant11("11225400"), qt.UserRole)
+	if err != nil {
+		fmt.Printf("Error setting UserData: %v\n", err)
+	}
+	rootItem.AppendChild(&item)
+
+	cellData2 := make([]*TreeCell, 0)
+	cellData2 = append(cellData2, NewTreeCell(qt.DisplayRole, *qt.NewQVariant11("Small File")))
+	cellData2 = append(cellData2, NewTreeCell(qt.DisplayRole, *qt.NewQVariant11("110 Bytes")))
+	item2 := NewTreeItem(cellData2, rootItem, false)
+	err = item2.SetData(1, qt.NewQVariant11("110"), qt.UserRole)
+	if err != nil {
+		fmt.Printf("Error setting UserData: %v\n", err)
+	}
+	rootItem.AppendChild(&item2)
+
+	cellData3 := make([]*TreeCell, 0)
+	cellData3 = append(cellData3, NewTreeCell(qt.DisplayRole, *qt.NewQVariant11("Medium File")))
+	cellData3 = append(cellData3, NewTreeCell(qt.DisplayRole, *qt.NewQVariant11("109.5 KiB")))
+	item3 := NewTreeItem(cellData3, rootItem, false)
+	err = item3.SetData(1, qt.NewQVariant11("112200"), qt.UserRole)
+	if err != nil {
+		fmt.Printf("Error setting UserData: %v\n", err)
+	}
+	rootItem.AppendChild(&item3)
+}
+
+func GetDialog() *qt.QDialog {
+	dialog := qt.NewQDialog3(nil, qt.Dialog)
+	dialog.SetWindowFlags(qt.WindowStaysOnTopHint)
+	dialogLayout := qt.NewQVBoxLayout2()
+	dialogLayout.SetSpacing(0)
+	dialogLayout.SetContentsMargins(0, 0, 0, 0)
+	dialog.SetContentsMargins(10, 10, 10, 10)
+	dialog.SetLayout(dialogLayout.Layout())
+	dialog.SetWindowTitle("Example Sort Model Usage")
+
+	treeView := GetTreeView(dialog.QWidget)
+
+	// Model
+	model, rootItem := GetModel(dialog.QObject)
+	PopulateData(rootItem)
+
+	sortModel := qt.NewQSortFilterProxyModel2(treeView.QObject)
+	sortModel.SetSortRole(int(qt.UserRole))
+	sortModel.OnLessThan(func(super func(source_left *qt.QModelIndex, source_right *qt.QModelIndex) bool, source_left *qt.QModelIndex, source_right *qt.QModelIndex) bool {
+		// In our case we know that the data from our TreeItem will come back as an int for our UserRole
+		leftData := sortModel.SourceModel().Data(source_left, sortModel.SortRole())
+		rightData := sortModel.SourceModel().Data(source_right, sortModel.SortRole())
+		return leftData.ToInt() < rightData.ToInt()
+	})
+
+	sortModel.SetSourceModel(model)
+	treeView.SetModel(sortModel.QAbstractItemModel)
+
+	dialogLayout.AddWidget3(treeView.QWidget, 0, qt.AlignCenter)
+	return dialog
```

Comment:
In the default view, the Size column is initially truncated, so I went to resize the window larger. But something about the `QSizePolicy` is meaning the treeview will not grow when the window is resized (although it does shrink). I think this could be fixed in a 1 line change, but I haven't checked exactly.

![image](https://github.com/user-attachments/assets/fcc8ef0e-c48e-46ea-bb8d-0855342f93b7)

![image](https://github.com/user-attachments/assets/c81ed966-ee01-4b89-8ef3-71154607beb1)

Reply:
So I took a deeper look at this.

I can't figure out how to make this work as I would expect.

The only way I can get it working is with the following hack:
```go
dialog.OnResizeEvent(func(super func(event *qt.QResizeEvent), event *qt.QResizeEvent) {
		dialog.ResizeWithQSize(event.Size())
                // Width minus dialog.ContentMargins Left/Right
		treeView.Resize(event.Size().Width()-20, treeView.Size().Height())
	})
```

But having written Qt before, obviously that is a messy hack.


Is there a possibility that the default Resize events aren't being propagated down to child widgets?